controllers/druid/handler.go

```diff
@@ -1302,30 +1302,54 @@ func makePodTemplate(nodeSpec *v1alpha1.DruidNodeSpec, m *v1alpha1.Druid, ls map
-		TopologySpreadConstraints: getTopologySpreadConstraints(nodeSpec),
-		Tolerations:               getTolerations(nodeSpec, m),
-		Affinity:                  getAffinity(nodeSpec, m),
-		ImagePullSecrets:          firstNonNilValue(nodeSpec.ImagePullSecrets, m.Spec.ImagePullSecrets).([]v1.LocalObjectReference),
-		Containers: []v1.Container{
-			{
-				Image:           firstNonEmptyStr(nodeSpec.Image, m.Spec.Image),
-				Name:            fmt.Sprintf("%s", nodeSpecUniqueStr),
-				Command:         []string{firstNonEmptyStr(m.Spec.StartScript, "bin/run-druid.sh"), nodeSpec.NodeType},
-				ImagePullPolicy: v1.PullPolicy(firstNonEmptyStr(string(nodeSpec.ImagePullPolicy), string(m.Spec.ImagePullPolicy))),
-				Ports:           nodeSpec.Ports,
-				Resources:       nodeSpec.Resources,
-				Env:             getEnv(nodeSpec, m, configMapSHA),
-				EnvFrom:         getEnvFrom(nodeSpec, m),
-				VolumeMounts:    getVolumeMounts(nodeSpec, m),
-				LivenessProbe:   getLivenessProbe(nodeSpec, m),
-				ReadinessProbe:  getReadinessProbe(nodeSpec, m),
-				StartupProbe:    getStartUpProbe(nodeSpec, m),
-				Lifecycle:       nodeSpec.Lifecycle,
-				SecurityContext: firstNonNilValue(nodeSpec.ContainerSecurityContext, m.Spec.ContainerSecurityContext).(*v1.SecurityContext),
-			},
+
+	var containers []v1.Container
+	containers = append(containers,
+		v1.Container{
+			Image:           firstNonEmptyStr(nodeSpec.Image, m.Spec.Image),
+			Name:            fmt.Sprintf("%s", nodeSpecUniqueStr),
+			Command:         []string{firstNonEmptyStr(m.Spec.StartScript, "bin/run-druid.sh"), nodeSpec.NodeType},
+			ImagePullPolicy: v1.PullPolicy(firstNonEmptyStr(string(nodeSpec.ImagePullPolicy), string(m.Spec.ImagePullPolicy))),
+			Ports:           nodeSpec.Ports,
+			Resources:       nodeSpec.Resources,
+			Env:             getEnv(nodeSpec, m, configMapSHA),
+			EnvFrom:         getEnvFrom(nodeSpec, m),
+			VolumeMounts:    getVolumeMounts(nodeSpec, m),
+			LivenessProbe:   getLivenessProbe(nodeSpec, m),
+			ReadinessProbe:  getReadinessProbe(nodeSpec, m),
+			StartupProbe:    getStartUpProbe(nodeSpec, m),
+			Lifecycle:       nodeSpec.Lifecycle,
+			SecurityContext: firstNonNilValue(nodeSpec.ContainerSecurityContext, m.Spec.ContainerSecurityContext).(*v1.SecurityContext),
 		},
+	)
+
+	if m.Spec.AdditionalContainer != nil {
+
+		for _, containerList := range m.Spec.AdditionalContainer {
+
+			containers = append(containers,
+				v1.Container{
+					Image:           containerList.Image,
+					Name:            containerList.ContainerName,
+					Resources:       containerList.Resources,
+					VolumeMounts:    containerList.VolumeMounts,
+					Command:         containerList.Command,
+					Args:            containerList.Args,
+					ImagePullPolicy: containerList.ImagePullPolicy,
+					SecurityContext: containerList.ContainerSecurityContext,
+					Env: 			 containerList.Env,
+				},
+			)
+		}
+	}
+
```

Comment:
This is absolute reference to pass in the value, in case the field is missing i expect the program to crash, ill suggest to test out the scenario and use pointers to reference, else writing functions and handling nil would be repetitive code.

Reply:
The current code is similar to how ports and resources are defined in the parent container
```
Ports:           nodeSpec.Ports,
Resources:       nodeSpec.Resources,
```

Reply:
@AdheipSingh : Let me know what you think.

Reply:
ill run locally and confirm the changes sometime by end of the week.
Thanks